From b0b8d138d6aa392bce27bd5cea3e6959a21aedb9 Mon Sep 17 00:00:00 2001
From: Bharathraj Nagaraju <snbraj@codeaurora.org>
Date: Thu, 20 Oct 2016 21:28:01 -0700
Subject: [PATCH] ARM: dts: msm: Add support for  msm8996 auto mctm.

Added device tree changes that are needed to support
MSM8996 auto mctm.

Change-Id: I47c6b877deead41c92795516518b9c12108e63c6
Signed-off-by: Bharathraj Nagaraju <snbraj@codeaurora.org>
---
 arch/arm/boot/dts/qcom/Makefile                 |   1 +
 arch/arm/boot/dts/qcom/msm8996pro-auto-mctm.dts | 426 ++++++++++++++++++++++++
 2 files changed, 427 insertions(+)
 create mode 100644 arch/arm/boot/dts/qcom/msm8996pro-auto-mctm.dts

diff --git a/arch/arm/boot/dts/qcom/Makefile b/arch/arm/boot/dts/qcom/Makefile
index 00e9842..237fa75 100644
--- a/arch/arm/boot/dts/qcom/Makefile
+++ b/arch/arm/boot/dts/qcom/Makefile
@@ -30,6 +30,7 @@ dtb-$(CONFIG_ARCH_MSM8996) += msm8996-v2-pmi8994-cdp.dtb \
 	msm8996-v3-pm8004-agave-adp.dtb \
 	msm8996-v3-pm8004-agave-adp-lite.dtb \
 	msm8996pro-auto-adp.dtb \
+	msm8996pro-auto-mctm.dtb \
 	msm8996pro-auto-adp-lite.dtb \
 	msm8996pro-auto-cdp.dtb \
 	msm8996pro-pmi8994-cdp.dtb \
diff --git a/arch/arm/boot/dts/qcom/msm8996pro-auto-mctm.dts b/arch/arm/boot/dts/qcom/msm8996pro-auto-mctm.dts
new file mode 100644
index 0000000..aecf776
--- /dev/null
+++ b/arch/arm/boot/dts/qcom/msm8996pro-auto-mctm.dts
@@ -0,0 +1,426 @@
+/* Copyright (c) 2015-2016, The Linux Foundation. All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 and
+ * only version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include "msm8996pro.dtsi"
+#include "msm8996-pm8994.dtsi"
+#include "msm8996-agave-adp.dtsi"
+#include "msm8996pro-auto.dtsi"
+
+/ {
+	model = "Qualcomm Technologies, Inc. MSM 8996 AUTO ADP MCTM";
+	compatible = "qcom,apq8096-adp", "qcom,msm8996", "qcom,adp";
+	qcom,msm-id = <315 0x10000>;
+	qcom,board-id = <0x05010019 5>;
+};
+
+&spi_9 {
+	status = "ok";
+	can-controller@0 {
+		compatible = "renesas,rh850";
+		reg = <0>;
+		interrupt-parent = <&tlmm>;
+		interrupts = <122 0>;
+		spi-max-frequency = <5000000>;
+	};
+};
+
+&soc {
+	qcom,msm-ssc-sensors {
+		status = "disabled";
+	};
+
+	qcom,msm-thermal {
+		qcom,hotplug-temp = <115>;
+		qcom,hotplug-temp-hysteresis = <25>;
+		qcom,therm-reset-temp = <119>;
+	};
+
+	qcom,adv7481@70 {
+		qcom,cam-vreg-min-voltage = <1300000 0 1800000>;
+		qcom,cam-vreg-max-voltage = <1300000 0 1800000>;
+	};
+
+	pcie0: qcom,pcie@00600000 {
+		perst-gpio = <&tlmm 30 0>;
+		qcom,vreg-cx-voltage-level = <7 5 0>;
+	};
+
+	qcom,msm-dai-tdm-quat-rx {
+		pinctrl-names = "default", "sleep";
+		pinctrl-0 = <&quat_mi2s_active &quat_mi2s_sd0_active
+				&quat_tdm_dout_active>;
+		pinctrl-1 = <&quat_mi2s_sleep &quat_mi2s_sd0_sleep
+				&quat_tdm_dout_sleep>;
+	};
+
+	tlmm: pinctrl@01010000 {
+		pcie0 {
+			pcie0_perst_default: pcie0_perst_default {
+				mux {
+					pins = "gpio30";
+					function = "gpio";
+				};
+
+				config {
+					pins = "gpio30";
+					drive-strength = <2>;
+					bias-pull-down;
+				};
+			};
+
+			ntn2_config: ntn2_config {
+				mux {
+					pins = "gpio139";
+					function = "gpio";
+				};
+
+				config {
+					pins = "gpio139";
+					bias-pull-up;		/* pull up */
+					drive-strength = <2>;	/* 2 MA */
+				};
+			};
+
+			ntn3_config: ntn3_config {
+				mux {
+					pins = "gpio140";
+					function = "gpio";
+				};
+
+				config {
+					pins = "gpio140";
+					bias-pull-up;		/* pull up */
+					drive-strength = <2>;	/* 2 MA */
+				};
+			};
+
+			ntn1_display_clk_sync: ntn1_display_clk_sync {
+				mux {
+					pins = "gpio12";
+					function = "gpio";
+				};
+
+				config {
+					pins = "gpio12";
+					bias-pull-down;                /* pull up */
+					drive-strength = <2>;     /* 2 MA */
+				};
+			};
+
+			ntn2_private_clk_sync: ntn2_private_clk_sync {
+				mux {
+					pins = "gpio13";
+					function = "gpio";
+				};
+
+				config {
+					pins = "gpio13";
+					bias-pull-down;                /* pull up */
+					drive-strength = <2>;     /* 2 MA */
+				};
+			};
+
+			ntn3_brain_clk_sync: ntn3_brain_clk_sync {
+				mux {
+					pins = "gpio69";
+					function = "gpio";
+				};
+
+				config {
+					pins = "gpio69";
+					bias-pull-down;                /* pull up */
+					drive-strength = <2>;     /* 2 MA */
+				};
+			};
+		};
+	};
+
+	qcom,ntn_avb@1 {
+		compatible = "qcom,ntn_avb";
+		ntn-rst-gpio = <&tlmm 139 0>;
+
+		qcom,ntn-rst-delay-msec = <100>;
+		qcom,ntn-rc-num = <0>;
+		qcom,ntn-bus-num = <4>; /* Private ethernet */
+		pinctrl-names = "default";
+		pinctrl-0 = <&ntn2_config>;
+	  };
+
+	qcom,ntn_avb@2 {
+		compatible = "qcom,ntn_avb";
+		ntn-rst-gpio = <&tlmm 140 0>;
+
+		qcom,ntn-rst-delay-msec = <100>;
+		qcom,ntn-rc-num = <0>;
+		qcom,ntn-bus-num = <5>; /* Display*/
+		pinctrl-names = "default";
+		pinctrl-0 = <&ntn3_config>;
+	};
+
+	qcom,ntn_avb@3 {
+		compatible = "qcom,ntn_avb";
+		ntn-rst-gpio = <&pm8994_gpios 13 0>;
+
+		vdd-ntn-hsic-supply = <&pm8994_l25>;
+		vdd-ntn-pci-supply = <&pm8994_s4>;
+		vdd-ntn-io-supply = <&pm8994_s4>;
+
+		qcom,ntn-rst-delay-msec = <100>;
+		qcom,ntn-rc-num = <1>;
+		qcom,ntn-bus-num = <1>;
+	};
+
+	i2c_1: i2c@7575000 {
+		compatible = "qcom,i2c-msm-v2";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg-names = "qup_phys_addr";
+		reg = <0x7575000 0x1000>;
+		interrupt-names = "qup_irq";
+		interrupts = <0 95 0>;
+		dmas = <&dma_blsp1 12 32 0x20000020 0x20>,
+				<&dma_blsp1 13 32 0x20000020 0x20>;
+		dma-names = "tx", "rx";
+		qcom,master-id = <86>;
+		qcom,clk-freq-out = <400000>;
+		qcom,clk-freq-in = <19200000>;
+		clock-names = "iface_clk", "core_clk";
+		clocks = <&clock_gcc clk_gcc_blsp1_ahb_clk>,
+				<&clock_gcc clk_gcc_blsp1_qup1_i2c_apps_clk>;
+		pinctrl-names = "i2c_active", "i2c_sleep";
+		pinctrl-0 = <&i2c_1_active>;
+		pinctrl-1 = <&i2c_1_sleep>;
+	};
+
+	qcom,msm-dai-tdm-tert-rx {
+		qcom,msm-cpudai-tdm-clk-rate = <12288000>;
+		dai_tert_tdm_rx_0: qcom,msm-dai-q6-tdm-tert-rx-0 {
+					qcom,msm-cpudai-tdm-sync-src = <1>;
+		};
+		dai_tert_tdm_rx_1: qcom,msm-dai-q6-tdm-tert-rx-1 {
+					qcom,msm-cpudai-tdm-sync-src = <1>;
+		};
+		dai_tert_tdm_rx_2: qcom,msm-dai-q6-tdm-tert-rx-2 {
+					qcom,msm-cpudai-tdm-sync-src = <1>;
+		};
+		dai_tert_tdm_rx_3: qcom,msm-dai-q6-tdm-tert-rx-3 {
+					qcom,msm-cpudai-tdm-sync-src = <1>;
+		};
+	};
+
+	qcom,msm-dai-tdm-tert-tx {
+		qcom,msm-cpudai-tdm-clk-rate = <12288000>;
+		dai_tert_tdm_tx_0: qcom,msm-dai-q6-tdm-tert-tx-0 {
+					qcom,msm-cpudai-tdm-sync-src = <1>;
+		};
+		dai_tert_tdm_tx_1: qcom,msm-dai-q6-tdm-tert-tx-1 {
+					qcom,msm-cpudai-tdm-sync-src = <1>;
+		};
+		dai_tert_tdm_tx_2: qcom,msm-dai-q6-tdm-tert-tx-2 {
+					qcom,msm-cpudai-tdm-sync-src = <1>;
+		};
+		dai_tert_tdm_tx_3: qcom,msm-dai-q6-tdm-tert-tx-3 {
+					qcom,msm-cpudai-tdm-sync-src = <1>;
+		};
+	};
+
+	qcom,msm-dai-tdm-quat-rx {
+		qcom,msm-cpudai-tdm-clk-rate = <12288000>;
+		dai_quat_tdm_rx_0: qcom,msm-dai-q6-tdm-quat-rx-0 {
+					qcom,msm-cpudai-tdm-sync-src = <1>;
+		};
+		dai_quat_tdm_rx_1: qcom,msm-dai-q6-tdm-quat-rx-1 {
+					qcom,msm-cpudai-tdm-sync-src = <1>;
+		};
+		dai_quat_tdm_rx_2: qcom,msm-dai-q6-tdm-quat-rx-2 {
+					qcom,msm-cpudai-tdm-sync-src = <1>;
+		};
+		dai_quat_tdm_rx_3: qcom,msm-dai-q6-tdm-quat-rx-3 {
+					qcom,msm-cpudai-tdm-sync-src = <1>;
+		};
+	};
+
+	qcom,msm-dai-tdm-quat-tx {
+		qcom,msm-cpudai-tdm-clk-rate = <12288000>;
+		dai_quat_tdm_tx_0: qcom,msm-dai-q6-tdm-quat-tx-0 {
+				qcom,msm-cpudai-tdm-sync-src = <1>;
+		};
+		dai_quat_tdm_tx_1: qcom,msm-dai-q6-tdm-quat-tx-1 {
+				qcom,msm-cpudai-tdm-sync-src = <1>;
+		};
+		dai_quat_tdm_tx_2: qcom,msm-dai-q6-tdm-quat-tx-2 {
+				qcom,msm-cpudai-tdm-sync-src = <1>;
+		};
+		dai_quat_tdm_tx_3: qcom,msm-dai-q6-tdm-quat-tx-3 {
+				qcom,msm-cpudai-tdm-sync-src = <1>;
+		};
+	};
+
+	i2c_10: i2c@75b8000 {
+		compatible = "qcom,i2c-msm-v2";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg-names = "qup_phys_addr";
+		reg = <0x75b8000 0x1000>;
+		interrupt-names = "qup_irq";
+		interrupts = <0 104 0>;
+		dmas = <&dma_blsp2 18 32 0x20000020 0x20>,
+				<&dma_blsp2 19 32 0x20000020 0x20>;
+		dma-names = "tx", "rx";
+		qcom,master-id = <84>;
+		qcom,clk-freq-out = <400000>;
+		qcom,clk-freq-in = <19200000>;
+		clock-names = "iface_clk", "core_clk";
+		clocks = <&clock_gcc clk_gcc_blsp2_ahb_clk>,
+				<&clock_gcc clk_gcc_blsp2_qup4_i2c_apps_clk>;
+		pinctrl-names = "i2c_active", "i2c_sleep";
+		pinctrl-0 = <&i2c_10_active>;
+		pinctrl-1 = <&i2c_10_sleep>;
+	};
+};
+
+&tlmm {
+	i2c_1 {
+		i2c_1_active: i2c_1_active {
+			mux {
+				pins = "gpio2", "gpio3";
+				function = "blsp_i2c1";
+			};
+
+			config {
+				pins = "gpio2", "gpio3";
+				drive-strength = <2>;
+				bias-disable;
+			};
+		};
+
+		i2c_1_sleep: i2c_1_sleep {
+			mux {
+				pins = "gpio2", "gpio3";
+				function = "blsp_i2c1";
+			};
+
+			config {
+				pins = "gpio2", "gpio3";
+				drive-strength = <2>;
+				bias-pull-down;
+			};
+		};
+	};
+
+	i2c_10 {
+		i2c_10_active: i2c_10_active {
+			mux {
+				pins = "gpio10", "gpio11";
+				function = "blsp_i2c10";
+			};
+
+			config {
+				pins = "gpio10", "gpio11";
+				drive-strength = <2>;
+				bias-disable;
+			};
+		};
+
+		i2c_10_sleep: i2c_10_sleep {
+			mux {
+				pins = "gpio10", "gpio11";
+				function = "blsp_i2c10";
+			};
+
+			config {
+				pins = "gpio10", "gpio11";
+				drive-strength = <2>;
+				bias-pull-down;
+			};
+		};
+	};
+};
+
+&i2c_1 {
+	ak4601@10 {
+		compatible = "akm,ak4601";
+		reg = <0x10>;
+		ak4601,pdn-gpio = <&tlmm 90 0>;/*FIXME*/
+		status = "ok";
+	};
+};
+
+
+&slim_msm {
+	status = "disabled";
+};
+
+&pil_modem {
+	pinctrl-names = "default";
+	pinctrl-0 = <&modem_mux>;
+};
+
+&pm8994_mpps {
+	mpp@a500 { /* MPP 6 */
+		qcom,mode = <1>;		/* Digital output */
+		qcom,output-type = <0>;		/* CMOS logic */
+		qcom,vin-sel = <2>;		/* S4 1.8V */
+		qcom,src-sel = <0>;		/* Constant */
+		qcom,master-en = <1>;		/* Enable GPIO */
+		status = "okay";
+	};
+};
+
+&sdhc_2 {
+	cd-gpios = <&tlmm 38 GPIO_ACTIVE_LOW>;
+	pinctrl-0 = <&sdc2_clk_on &sdc2_cmd_on &sdc2_data_on &sdc2_cd_on_sbc>;
+	pinctrl-1 = <&sdc2_clk_off &sdc2_cmd_off &sdc2_data_off
+		&sdc2_cd_on_sbc>;
+};
+
+&i2c_7 {
+	silabs4705@11 { /* SiLabs FM chip, slave id 0x11*/
+		status = "disabled";
+	};
+};
+
+/ {
+	aliases {
+		i2c1 = &i2c_1;
+		i2c10 = &i2c_10;
+	};
+};
+
+&spi_0 {
+	status = "disabled";
+};
+
+&spi_10 {
+	status = "disabled";
+};
+
+&i2c_7 {
+	status = "disabled";
+};
+
+&i2c_8 {
+	status = "disabled";
+};
+
+&spi_9 {
+	status = "ok";
+	can-controller@0 {
+		compatible = "nxp,mpc5745c";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+	};
+};
+
-- 
2.9.0.137.gcf4c2cf

